pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -70,6 +70,16 @@ type InstallerController struct {
 	installerPodImageFn func() string
 	// ownerRefsFn sets the ownerrefs on the pruner pod
 	ownerRefsFn func(revision int32) ([]metav1.OwnerReference, error)
+
+	installerPodMutationFn InstallerPodMutationFunc
+}
+
+// InstallerPodMutationFunc is a function that has a chance at changing the installer pod before it is created
+type InstallerPodMutationFunc func(pod *corev1.Pod, nodeName string, operatorSpec *operatorv1.StaticPodOperatorSpec, revision int32) error
+
+func (o *InstallerController) WithInstallerPodMutationFn(installerPodMutationFn InstallerPodMutationFunc) *InstallerController {
```

Comment:
func sounds like those funcs can be chained by calling it multiple times.

Reply:
sure, you prefer an appending slice of mutators?